emcc.py

```diff
@@ -460,18 +460,13 @@ def is_valid_abspath(path_name):
       def check_bad_eq(arg):
         assert '=' not in arg, 'Invalid parameter (do not use "=" with "--" options)'
 
-      # Defaults to not showing absolute path warnings
-      absolute_warning_shown = True
+      # Scan and strip emscripten specific cmdline warning flags
+      # This needs to run before other cmdline flags have been parsed, so that warnings are properly printed during arg parse
+      newargs = shared.WarningManager.capture_warnings(newargs)
+      absolute_warning_shown = False
```

Comment:
could the warning manager perhaps also track if the warning was shown? if it handled printing the warnings itself, then it could handle showing each at most once.